higher-order-functions/exercise.js

```diff
@@ -45,7 +68,18 @@ function union(arrays) {}
 // should log: [5, 10, 15, 88, 1, 7, 100]
 
 //Extension 5
-function objOfMatches(array1, array2, callback) {}
+function callback(array1,array2){
+    const obj = {};
+    for(let i = 0; i < array1.length; i++){
+        if(array1[i] == array2[i]){
+            obj[array1[i]] = array2[i];
+        }
+    }
+    return obj;
+}
+function objOfMatches(array1, array2, callback) {
+    return callback(array1,array2);
+}
```

Comment:
This will not work you are defining two function with the same name `callback`.

```js
function objOfMatches(array1, array2, callback){
 return array1.reduce((acc, value, i) => {
    if(callback(value) === array2[i]){
      acc[value] = array2[i];
    }
  return acc;
 },{})
}
```